docs/src/external.md

```diff
@@ -0,0 +1,47 @@
+# External libraries and packages
+
+Other libraries and packages may also make use of MPI. There are several concerns to ensure things are set up correctly.
+
+## Binary requirements
+
+You will need to ensure that the binaries are built correctly. In particular, if you are using [using a system-provided MPI backend](@ref using_system_mpi), you will also need to use the system provided binary for the package in question.
+
+## Passing handles via `ccall`
+
+When passing MPI.jl handle objects ([`MPI.Comm`](@ref), [`MPI.Info`](@ref), etc) to C/C++ functions [via `ccall`](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/), you should pass the object directly as an argument, and specify the argument type as either the underlying handle type (`MPI.API.MPI_Comm`, `MPI.API.MPI_Info`, etc.), or a pointer (`Ptr{MPI.API.MPI_Comm}`, `Ptr{MPI.API.MPI_Info}`, etc.). This will internally handle the unwrapping, but ensure that a reference is kept to avoid premature garbage collection.
```

Comment:
```suggestion
When passing MPI.jl handle objects ([`MPI.Comm`](@ref), [`MPI.Info`](@ref), etc) to C/C++ functions [via `ccall`](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/), you should pass the object directly as an argument, and specify the argument type as either the underlying handle type (`MPI.MPI_Comm`, `MPI.API.MPI_Info`, etc.), or a pointer (`Ptr{MPI.MPI_Comm}`, `Ptr{MPI.MPI_Info}`, etc.). This will internally handle the unwrapping, but ensure that a reference is kept to avoid premature garbage collection.
```
In my experience it works well without the `API` part. If we require the `API` part for sanity/future compatibility, I think we should state it somewhere prominently which parts of MPI.jl are expected to be called via `API` and which are just `MPI.xxx`. If there's no hard requirement yet, I'd prefer to not use the `API` part to avoid confusion. If it's just that some commonly used types are used via `MPI.xxx` and other, less commonly used or low-level entities are available only via `MPI.API.yyy` it should be documented as well.

Reply:
@vchuravy @giordano what do you think?

Reply:
Probably I agree it's nice to keep `MPI_Comm`  part of the "high-level" API (i.e. within `MPI`, not `MPI.API`).